src/operator/ndarray_op.cc

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright (c) 2015 by Contributors
+ * \file ndarray_op.cc
+ * \brief
+ * \author Junyuan Xie
+*/
+#include "./ndarray_op-inl.h"
+#include <mxnet/ndarray.h>
+
+namespace mxnet {
+namespace op {
+template<>
+Operator *CreateOp<cpu>(NDArrayOpParam param) {
+  return new NDArrayOp<cpu>(param);
+}
+
+template<>
+Context NDArrayOp<cpu>::get_ctx() {
+  return Context::CPU();
+}
+
+template<typename xpu>
+void NDArrayOp<xpu>::Forward(const OpContext &ctx,
+                   const std::vector<TBlob> &in_data,
+                   const std::vector<OpReqType> &req,
+                   const std::vector<TBlob> &out_data,
+                   const std::vector<TBlob> &aux_args) {
+  using namespace mshadow;
+  Context ndctx = get_ctx();
+  std::vector<void*> ptrs;
+  std::vector<int> tags;
+  for (auto& i : req) CHECK_NE(i, kAddTo);
+  APPEND(in_data, 0);
+  APPEND(out_data, 1);
+  param_.pinfo->forward(ptrs.size(), ptrs.data(), tags.data(), param_.pinfo->p_forward);
+}
+
+template<typename xpu>
+void NDArrayOp<xpu>::Backward(const OpContext &ctx,
+                    const std::vector<TBlob> &out_grad,
+                    const std::vector<TBlob> &in_data,
+                    const std::vector<TBlob> &out_data,
+                    const std::vector<OpReqType> &req,
+                    const std::vector<TBlob> &in_grad,
+                    const std::vector<TBlob> &aux_args) {
+  using namespace mshadow;
+  Context ndctx = get_ctx();
+  std::vector<void*> ptrs;
+  std::vector<int> tags;
+  for (auto& i : req) CHECK_NE(i, kAddTo);
```

Comment:
This is fine as long as we do not use CUDA code completely, in this example, everything can actually be compiled with cc file